module/icp/algs/blake3/blake3_impl.c

```diff
@@ -28,6 +28,11 @@
 
 #include "blake3_impl.h"
 
+#ifdef __APPLE__
+/* Sadly, toolchain sets this, but M1 don't have normal NEON */
```

Comment:
NEON is used in [AOM](https://aomedia.googlesource.com/aom), and way it's used could be derived from it. I found that git version of this library in QLVideo main has problems with NEON (few compiler options need to be added). Later version (in one of PRs) doesn't have any problems with it

Reply:
BTW, [Blake3](https://github.com/BLAKE3-team/BLAKE3) has direct support for macOS/aarch64

`ARM_NEON` definitions can be found `<simd/common.h>`

Reply:
Awesome, thanks. With assembler, I've had to make an osx/ copy due to different sets of scratch registers available - have ZOL looked at a way around that?   At the moment, I can't easily test NEON, since M1/macos does not yet run as a VM client.